consensus/ethash/consensus_test.go

```diff
@@ -71,9 +73,12 @@ func TestCalcDifficulty(t *testing.T) {
 	}
 
 	config := &params.ChainConfig{HomesteadBlock: big.NewInt(1150000)}
+	chain, _ := core.NewBlockChain(nil, config, nil, vm.Config{})
+	defer chain.Stop()
+
 	for name, test := range tests {
 		number := new(big.Int).Sub(test.CurrentBlocknumber, big.NewInt(1))
-		diff := CalcDifficulty(config, test.CurrentTimestamp, &types.Header{
+		diff := NewFaker().CalcDifficulty(chain, test.CurrentTimestamp, &types.Header{
```

Comment:
If you retain the global `ethash.CalcDifficulty` method too, you won't need either the NewFaker, nor the emphemeral chain here.